pandas/core/generic.py

```diff
@@ -2530,6 +2535,14 @@ def to_json(
             date_format = "iso"
         elif date_format is None:
```

Comment:
I think need to warn for anything that is not currently `iso`, including when date_format is None (although the message will be different)

Reply:
Can you also add a warning for the `date_format=None` case that previously defaulted to "epoch"; this should in the future default to "iso"

Reply:
Most of the `to_json` use cases don't involve dates and wouldn't be affected by the `date_format` value, throwing a warning in these cases might be unnecessary, essentially they will need to pass `date_format='iso'` for no reason to silence this warning, are you sure we should do this?

Reply:
Sorry to be more specific we need to warn when `date_format=None` and we actually serialize timestamp types. I agree no point in warning if a DataFrame has no timestamp type, but if users are relying on the default epoch behavior they need to be warned of the change

Reply:
@WillAyd curious how would users get the old behavior? It would be good to add that in the warning message

Reply:
The old behavior as in just an integer? I think the problem with that is it was an implementation detail of pandas spilling out into the JSON serializer. Historically our timestamps were exclusively nanoseconds since the Unix epoch, but with all the work @jbrockmendel has been doing that is no longer true (and _usually not true).

Reply:
> The old behavior as in just an integer?

Yeah. Just checking if we can still offer a suggestion for a migration path if they want to keep the old behavior

Reply:
I don't think so. Especially with our auto-inferencing of resolutions I don't see how it would be usable at all roundtripping through JSON

Reply:
OK sounds good